vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ReferenceManagerTest.java

```diff
@@ -0,0 +1,339 @@
+
+        ref1 = new Reference();
+        ref1.setId(REF_ID1);
+        ref1.setTitle(REF_TITLE1);
+        ref1.setAuthor(REF_AUTHOR1);
+        ref1.setYear(REF_YEAR1);
+        ref1.setJournal(REF_JOURNAL1);
+        ref1.setUrl(REF_URL1);
+        ref1.setVolume(REF_VOLUME1);
+        ref1.setIssue(REF_ISSUE1);
+        ref1.setPages(REF_PAGES1);
+        ref1.setEditors(REF_EDITORS1);
+        ref1.setType(REF_TYPE1);
+        ref1.setNote(REF_NOTE1);
+
+        ref2 = new Reference();
+        ref2.setId(REF_ID2);
+        ref2.setTitle(REF_TITLE2);
+        ref2.setAuthor(REF_AUTHOR2);
+        ref2.setYear(REF_YEAR2);
+        ref2.setJournal(REF_JOURNAL2);
+        ref2.setUrl(REF_URL2);
+        ref2.setVolume(REF_VOLUME2);
+        ref2.setIssue(REF_ISSUE2);
+        ref2.setPages(REF_PAGES2);
+        ref2.setEditors(REF_EDITORS2);
+        ref2.setType(REF_TYPE2);
+        ref2.setNote(REF_NOTE2);
+
+        refList = new ArrayList<>();
+        refList.add((Reference) ref2);
+        refList.add((Reference) ref1);
+
+        refForm = new Reference();
+        refForm.setTitle(REF_NEW_TITLE);
+        refForm.setAuthor(REF_NEW_AUTHOR);
+        refForm.setYear(REF_NEW_YEAR);
+        refForm.setJournal(REF_NEW_JOURNAL);
+        refForm.setUrl(REF_NEW_URL);
+        refForm.setVolume(REF_NEW_VOLUME);
+        refForm.setIssue(REF_NEW_ISSUE);
+        refForm.setPages(REF_NEW_PAGES);
+        refForm.setEditors(REF_NEW_EDITORS);
+        refForm.setType(REF_NEW_TYPE);
+        refForm.setNote(REF_NEW_NOTE);
+
+        ReflectionTestUtils.setField(refManagerToTest, "pageSize", 10);
+    }
+
+    @Test
+    public void test_deleteReferenceById_refIdPresent() throws ReferenceDoesNotExistException {
+        Reference ref = new Reference();
+        ref.setId(REF_ID1);
+        Optional<Reference> refObj = Optional.of(ref);
+        Mockito.when(refRepo.findById(ref.getId())).thenReturn(refObj);
+        refManagerToTest.deleteReferenceById(REF_ID1);
+        Mockito.verify(refRepo).deleteById(REF_ID1);
+    }
+
+    @Test
+    public void test_deleteReferenceById_refIdNotPresent() throws ReferenceDoesNotExistException {
+        Mockito.when(refRepo.findById(REF_ID_NOT_PRESENT)).thenReturn(Optional.empty());
+        refManagerToTest.deleteReferenceById(REF_ID_NOT_PRESENT);
+        Mockito.verify(refRepo, Mockito.never()).deleteById(REF_ID_NOT_PRESENT);
+    }
+
+    @Test
+    public void test_deleteReferenceId_refIdIsNull() throws ReferenceDoesNotExistException {
+        Mockito.when(refRepo.findById(null)).thenReturn(Optional.empty());
+        refManagerToTest.deleteReferenceById(null);
+        Mockito.verify(refRepo, Mockito.never()).deleteById(REF_ID_NOT_PRESENT);
+    }
+
+    @Test
+    public void test_getTotalPages_success() {
+        when(refRepo.count()).thenReturn(12L);
+        assertEquals(2, refManagerToTest.getTotalPages());
+    }
+
+    @Test
+    public void test_getTotalPages_whenZeroImages() {

```

Comment:
what does 'whenZeroImages' mean?  These are tests for references, arent' they?

Reply:
Yes corrected now.